### PR TITLE
chore(flags): add cypress regression test to ensure flags rendering as expected in FlagSelector component

### PR DESCRIPTION
**Pull request by havenbarnes**
## Problem

Adding a regression test to catch breakages like the one that was fixed here: https://github.com/PostHog/posthog/pull/26099

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Ran cypress locally

### CODE REVIEW

File: frontend/src/scenes/surveys/SurveyEdit.tsx [hunk 1]
```diff
@@ -511,7 +511,7 @@ export default function SurveyEdit(): JSX.Element {
                             : []),
                         {
                             key: SurveyEditSection.DisplayConditions,
-                            header: 'Display conditions',
+                            header: <div data-attr="survey-display-conditions">Display conditions</div>,
```
Comment (havenbarnes) [inline frontend/src/scenes/surveys/SurveyEdit.tsx:514]: Not sure if there's a more preferable option for targeting this - the div wrap has no visual effect. Could go with a more brittle cypress `contains('Display conditions')` instead, or add formal support for `id`'s or `data-attr`'s in `LemonCollapse`

Comment (dmarticus) [reply frontend/src/scenes/surveys/SurveyEdit.tsx:514]: Yeah, this is fine, if a bit ugly – even thought `header` can be a ReactNode, we basically just pass it strings every time.  Would be cool if `LemonCollapse` support `data-attrs` instead; how hard would that be?

**Review by dmarticus — COMMENTED** [review]
this is generally fine, but would prefer we push the `data-attr` upstream to `LemonCollapse` if possible.

**Review by dmarticus — APPROVED** [review]
this is generally fine, but would prefer we push the `data-attr` upstream to `LemonCollapse` if possible.

File: frontend/src/scenes/surveys/SurveyEdit.tsx (hunk 1 above)
Comment (havenbarnes) [reply frontend/src/scenes/surveys/SurveyEdit.tsx:514]: Yup, not too bad to do that, sounds good to me

Comment (havenbarnes) [reply frontend/src/scenes/surveys/SurveyEdit.tsx]: done!

**Pull request merged**